### PR TITLE
prov/shm: ptrace check and adjustment, fabtests/common: avoid testing unsupported sizes

### PR DESCRIPTION
**Pull request by aingerson**

### CODE REVIEW

**Comment by aingerson** [comment]
@shefty @rajachan Short term work around we were discussing
Let me know if there's a simpler way to check the value rather than opening the file

File: prov/shm/src/smr_init.c [hunk 1]
```diff
@@ -80,6 +81,23 @@ static int smr_getinfo(uint32_t version, const char *node, const char *service,
...
+				"Error getting value from ptrace_scope\n");
+			return -FI_EINVAL;
+		}
+		ret = fclose(file);
+		if (ret) {
+			FI_WARN(&smr_prov, FI_LOG_CORE,
+				"Error closing ptrace_scope file\n");
+			return -FI_EINVAL;
+		}
+	}
```
Comment (shefty) [inline prov/shm/src/smr_init.c]: Can you move this check into a function, e.g. smr_ptrace_enabled()?

File: prov/shm/src/smr_init.c [hunk 2]
```diff
@@ -100,6 +118,8 @@ static int smr_getinfo(uint32_t version, const char *node, const char *service,
 			cur->ep_attr->max_order_waw_size = 0;
 			cur->ep_attr->max_order_war_size = 0;
 		}
+		if (ptrace)
+			cur->ep_attr->max_msg_size = SMR_INJECT_SIZE;
```
Comment (shefty) [inline prov/shm/src/smr_init.c]: The use of ptrace as a bool here looks opposite from what the code is doing.  ptrace is true, so don't use it?

Comment (rajachan) [reply prov/shm/src/smr_init.c]: Yeah, CMA requires ptrace so this is ambiguous. Maybe call it ptrace_scope and  fallback if it is at a non-zero level, or just move it to a boolean function like Sean suggests and fallback if the scope is restrictive.

**Comment by aingerson** [comment]
Updated ptrace patch, including commit message to clarify

File: prov/shm/src/smr_init.c [hunk 3]
```diff
@@ -61,14 +61,38 @@ static void smr_resolve_addr(const char *node, const char *service,
 	*addrlen = strlen(*addr);
 }
 
+static int smr_get_ptrace_scope(void)
+{
+	FILE *file;
+	int scope, ret;
+
+	scope = 0;
```
Comment (shefty) [inline prov/shm/src/smr_init.c]: Maybe initialize to ENOSYS/ENOTSUPP and only return 0 if we can verify that scope is 0?

Comment (aingerson) [reply prov/shm/src/smr_init.c]: Well the file doesn't always exist. If it doesn't, I believe it defaults to 0. At least it does on our rhel systems. Perhaps there's another way to verify. I wasn't able to find another way.

File: fabtests/include/shared.h [hunk 4]
```diff
@@ -284,6 +277,13 @@ char *cnt_str(char str[FT_STR_LEN], long long cnt);
 int size_to_count(int size);
 size_t datatype_to_size(enum fi_datatype datatype);
 
+static inline int ft_use_size(int index, int enable_flags)
+{
+	return test_size[index].size <= fi->ep_attr->max_msg_size &&
+		((enable_flags == FT_ENABLE_ALL) ||
+		(enable_flags & test_size[index].enable_flags));
+}
```
Comment (shefty) [inline fabtests/include/shared.h]: The dgram_pingpong test can be updated to remove its check for max_msg_size and rely on the above check instead.

File: prov/shm/src/smr_init.c (hunk 3 above)
Comment (shefty) [reply prov/shm/src/smr_init.c]: All righty then.  :)

File: fabtests/include/shared.h (hunk 4 above)
Comment (aingerson) [reply fabtests/include/shared.h]: Good call. Updated

**Pull request merged**